src/clients/ApplicationClient.js

```diff
@@ -704,7 +704,130 @@ export default class ApplicationClient extends BaseClient {
+   * Finding a resource group
+   * Querying a resource group
+   * Creating and deleting a resource group
+   * Updating group properties
+   * Retrieving and updating device properties
+   * 
+   */
+  
+	//getGatewayGroup(gatewayId)
+	//updateDeviceRoles(deviceId, roles[])
+	//getAllDevicesInGropu(groupId)
+	//addDevicesToGroup(groupId, deviceList[])
+	//removeDevicesFromGroup(groupId, deviceList[])
+  
+  getGatewayGroup(gatewayId){
+    this.log.debug("[ApplicationClient] getGatewayGroup("+gatewayId+")");
+    return this.callApi('GET', 200, true, ['authorization', 'devices' , gatewayId], null);
+  }
+  
+  updateDeviceRoles(deviceId, roles){
+    this.log.debug("[ApplicationClient] updateDeviceRoles("+deviceId+","+roles+")");
+    return this.callApi('PUT', 200, false, ['authorization', 'devices', deviceId, 'roles'], roles);
+  }  
+
+  getAllDevicesInGroup(groupId){
+    this.log.debug("[ApplicationClient] getAllDevicesInGropu("+groupId+")");
+    return this.callApi('GET', 200, true, ['bulk', 'devices' , groupId], null);
+  }
+
+  addDevicesToGroup(groupId, deviceList){
+    this.log.debug("[ApplicationClient] addDevicesToGroup("+groupId+","+deviceList+")");
+    return this.callApi('PUT', 200, false, ['bulk', 'devices' , groupId, "add"], deviceList);
+  }
+
+  removeDevicesFromGroup(groupId, deviceList){
+    this.log.debug("[ApplicationClient] removeDevicesFromGroup("+groupId+","+deviceList+")");
+    return this.callApi('PUT', 200, false, ['bulk', 'devices' , groupId, "remove"], deviceList);
+  }
+
+  // https://console.ng.bluemix.net/docs/services/IoT/gateways/gateway-access-control.html
+  
+	// Finding a Resource Group
+		// getGatewayGroups()
+	// Querying a resource group
+		// getUniqueDevicesInGroup(groupId)
+		// getUniqueGatewayGroup(groupId)
+	// Creating and deleting a resource group
+		// createGatewayGroup(groupName)
+		// deleteGatewayGroup(groupId)
+	// Retrieving and updating device properties
+		// getGatewayGroupProperties()
+		// getDeviceRoles(deviceId)
+		// updateGatewayProperties(gatewayId,control_props)
+		// updateDeviceControlProperties(deviceId, withroles)
+
+  // Finding a Resource Group
+  getGatewayGroups(){
+    this.log.debug("[ApplicationClient] getGatewayGroups()");
+    return this.callApi('GET', 200, true, ['groups'], null);  
+   }
+   
+  // Querying a resource group
+
+  // Get unique identifiers of the members of the resource group
+  getUniqueDevicesInGroup(groupId){
+    this.log.debug("[ApplicationClient] getUniqueDevicesInGroup("+groupId+")");
+    return this.callApi('GET', 200, true, ['bulk', 'devices' , groupId, "ids"], null);
+  }
+
+  // properties of the resource group
+  getUniqueGatewayGroup(groupId){
+    this.log.debug("[ApplicationClient] getUniqueGatewayGroup("+groupId+")");
+    return this.callApi('GET', 200, true, ['groups', groupId], null);
+  } 
+  
+  // Creating and deleting a resource group
+
+  // Create a Resource Group
+  createGatewayGroup(groupName){
```

Comment:
Is this a valid API?  I know there is a `POST /groups` API listed

If that's what was intended, let's just call this `createGroup(groupInfo)`